File: 190_reverse_bits/v_a.py
class Solution:
    def reverseBits(self, n: int) -> int:
        constructed = 0;
        for _ in range(31):
            bit = n & 1;
            n = n >> 1;
            constructed += bit
            constructed = constructed << 1
        bit = n & 1;
        constructed += bit
        return constructed;

File: 190_reverse_bits/v_a.c
int reverseBits(int n) {
    unsigned char bit = 0;
    int constructed = 0;
    for(size_t i = 0; i < 31; ++i){
        bit = n & 1;
        n >>= 1;
        constructed += bit;
        constructed <<= 1;
    }
    bit = n & 1;
    constructed += bit;
    
    return constructed;
}